common/main/java/com/couchbase/lite/internal/core/C4Database.java

```diff
@@ -250,15 +247,13 @@ static C4Database getDatabase(
                 "Parent directory does not exist or is not a directory: " + parentDirPath);
         }
 
-        if (VERSION_VECTORS_ENABLED) { flags |= C4Constants.DatabaseFlags.VERSION_VECTORS; }
-
         return new ManagedC4Database(
             impl,
             name,
             impl.nOpen(
                 parentDirPath,
                 name,
-                flags,
+                flags | C4Constants.DatabaseFlags.VERSION_VECTORS,
```

Comment:
This seems odd to me to tack this on here.  What is the flow where this is expected to be missing?

Reply:
If version vectors is on, it is always on.  All databases get created with the version vectors flag.  Putting it here guarantees that.  If it were not here, it would have to be in every place that called this method.  That would not make sense.
There is no other flag that works this way.  There are tests here that do not apply the CREATE flag; that do apply the READ_ONLY flag.  There used to be tests that applied the FAKE_CLOCK and production code that did not.  The VERSION_VECTOR flag is the only one that either must always be on, or always be off.

Your comment makes me think I should get rid of the DB_FLAGS constant.  It is confusing.  I'm going to create a ticket for myself, to do that.  Looking at the code, there is no reason for most of these methods to pass a flags argument around at all.